src/Directory.BeforeCommon.targets

```diff
@@ -110,7 +110,7 @@
     <DefineConstants Condition="$([MSBuild]::IsOSPlatform('windows'))">$(DefineConstants);TEST_ISWINDOWS</DefineConstants>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(MonoBuild)' != 'true' and ($([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp1.0')) or $(TargetFramework.StartsWith('netstandard')))">
+  <PropertyGroup Condition="'$(MonoBuild)' != 'true' and ($([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETCoreApp' or $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETStandard')">
```

Comment:
Maybe save off $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)') so you don't have to call it five times?

Also, you could shorten the Build.targets one to ... == '.NETFramework' and ... != '.NETFramework'

Reply:
NTS from PR review before I forget: Check a binlog file to see which file gets imported first. That's where to set the <TargetFrameworkIdentifier> property.

Reply:
Or put a conditionally included property in each. Probably safer since you wouldn't test every possible scenario.